R/publish.R

```diff
@@ -1,49 +1,44 @@
-#' Publish a book to the web
-#'
-#' Publish a book to the web. Note that you should be sure to render all
-#' versions of the book before publishing, unless you have specified
-#' \code{render = TRUE}.
-#'
-#' @inheritParams rsconnect::deploySite
+#' Publish a book to a Posit Connect server
 #'
+#' Publish a book to a Connect Server. By default, you should render the book
+#' locally before publishing.
 #' @param name Name of the book (this will be used in the URL path of the
 #'   published book). Defaults to the \code{book_filename} in
 #'   \code{_bookdown.yml} if not specified.
 #' @param account Account name to publish to. Will default to any previously
 #'   published to account or any single account already associated with
 #'   \code{server}.
-#' @param server Server to publish to (by default beta.rstudioconnect.com but
-#'   any RStudio Connect server can be published to).
-#'
+#' @param server Server to publish to (by default connect.posit.cloud, but any
+#'   Posit Connect server can be published to).
+#' @param ... Other arguments to be passed to [rsconnect::deploySite()].
+#' @note Previously the default server was bookdown.org, which will be sunset.
+#'   You are no longer recommended to publish to bookdown.org.
 #' @export
-publish_book = function(
-  name = NULL, account = NULL, server = NULL, render = c("none", "local", "server")
-) {
-
-  # if there are no RS Connect accounts setup on this machine
-  # then offer to add one for bookdown.org
-  accounts <- rsconnect::accounts()
-  accounts <- subset(accounts, server != "shinyapps.io")
-  if (is.null(accounts) || nrow(accounts) == 0) {
-
-    # add the server if we need to
-    servers = rsconnect::servers()
-    if (nrow(subset(servers, name == 'bookdown.org')) == 0)
-      rsconnect::addServer("https://bookdown.org/__api__", 'bookdown.org')
-
-    # see if they want to configure an account (bail if they don't)
-    message('You do not currently have a bookdown.org publishing account ',
-            'configured on this system.')
-    result = readline('Would you like to configure one now? [Y/n]: ')
-    if (tolower(result) == 'n') return(invisible())
-
-    # configure the account
-    rsconnect::connectUser(server = 'bookdown.org')
+publish_book = function(name = NULL, account = NULL, server = "connect.posit.cloud", ...) {
```

Comment:
I'm not an expert on how `rsconnect` works, but IIUC, the old code was relying on the `NULL` behavior to pick an account/credentials in the case where there is only one server configured in your environment. By making `server` non-NULL here, we would break the behavior for anyone who is publishing bookdown to a Posit Connect server like that.

Reply:
Another good point. Amended via e11cbead995894e6f007abf132e9775be81b4d8d.